cdap-data-fabric/src/main/java/co/cask/cdap/data2/util/hbase/HBaseTableUtil.java

```diff
@@ -385,6 +388,73 @@ public OutputStream getOutput() throws IOException {
     return info;
   }
 
+  protected String prefixCDAPToNamespace(String namespace) {
+    return "cdap_" + namespace;
+  }
+
+  /**
+   * Returns if the current version of HBase supports namespaces
+   *
+   * @return true if namespaces are supported, false otherwise
+   */
+  public abstract boolean namespacesSupported();
+
+  /**
+   * Creates a new {@link HTable} which may contain an HBase namespace depending on the HBase version
+   *
+   * @param conf the hadoop configuration
+   * @param namespace the namespace in which the table must exist
+   * @param tableName the name of the table to create a descriptor for
+   * @return an {@link HTable} for the tableName in the namespace
+   */
+  public abstract HTable getHTable(Configuration conf, @Nullable String namespace, String tableName) throws IOException;
```

Comment:
I agree with the idea. Is ``Id.Dataset`` better? It could potentially contain more things later but for now has ``Id.Namespace`` and a String ``tableName``?